### DISCUSSION TITLE
Find 'include' files from *not* meson project

### DISCUSSION THREAD

**Discussion by MartinSchwarzbauer** [mesonbuild/meson#12815] [Q&A] [answered]
Hi *!

We have the following situation.
I have to find some *include* files from a "parallel project" => linux drivers! This project is not a "meson project".

I tried the following solution:
```

# find driver
compiler = meson.get_compiler('c')
driver_inc = ''
if compiler.has_header('drv/myheader.h')
    message('Driver found in sysroot')
elif compiler.has_header('drv/myheader.h', include_directories: include_directories('../driver/inc') )
    driver_inc = '../driver/inc'
    message('Driver found in dependencies/driver/inc')
else
    assert(false, 'Driver not found')
endif

```

This worked as expected ---- **BUT** when i use this project as a dependency to an other project I get a lot of info and warnings:

```
meson.build:67: WARNING: include_directories sandbox violation!
The project is trying to access the directory './dependencies/driver/inc' which belongs to a different
subproject. This is a problem as it hardcodes the relative paths of these two projects.
This makes it impossible to compile the project in any other directory layout and also
prevents the subproject from changing its own directory layout.

Instead of poking directly at the internals the subproject should be executed and
it should set a variable that the caller can then use. Something like:

```

How can I use the includes from local project, also when this one is used as a subproject? I wanna fix this warning, because in future releases this will be an error!

Thank you so much,
Martin

**Comment by eli-schwartz** [comment, marked as answer]
It sounds like your project structure is:

```
.
├── drivers-project
└── myproject
```

And potentially:
```
.
└── otherproject
    └── subprojects
        ├── drivers-project
        └── myproject
```

I would suggest making drivers-project into a "wrapped" meson project. Give it a really simple meson.build that simply produces a
```meson
drivers_dep = declare_dependency(include_directories: ['inc'])
```

and you can get it with `drivers_dep = subproject('drivers-project').get_variable('drivers_dep')` and use it as a dependency in `elif compiler.has_header('drv/myheader.h', dependencies: drivers_dep)`.

**Comment by MartinSchwarzbauer** [comment]
Thank you very much! It's working for our project!
